graal-js/src/com.oracle.truffle.js.test/src/com/oracle/truffle/js/test/interop/resources/moduleblock/benchmark/kMeansModuleModuleBlock.js

```diff
@@ -0,0 +1,148 @@
+ *
+ * Subject to the condition set forth below, permission is hereby granted to any
+ * person obtaining a copy of this software, associated documentation and/or
+ * data (collectively the "Software"), free of charge and under any and all
+ * copyright rights in the Software, and any and all patent rights owned or
+ * freely licensable by each licensor hereunder covering either (i) the
+ * unmodified Software as contributed to or provided by such licensor, or (ii)
+ * the Larger Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ *
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ *
+ * The above copyright notice and either this complete permission notice or at a
+ * minimum a reference to the UPL must be included in all copies or substantial
+ * portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+export var moduleBlock = (async function() {
+
+    var kMeans = module {
+
+        const N = 100000; // datapoints
+        const K = 7; // cluster
+        const SIZE = 600 // max data size
+
+        const N_EXP = 20; // 20 runs
+
+        const clusterX = new Array(K);
+        const clusterY = new Array(K);
+
+        const dataPointsX = new Array(N);
+        const dataPointsY = new Array(N);
+        const dataPointsCluster = new Array(N);
+
+        function computeDist(x1, y1, x2, y2) {
+            let dx = x1-x2;
+            let dy = y1-y2;
+
+            return dx * dx + dy * dy;
+        }
+
+        function getClosestCluster(x,y) {
+            let minCluster=-1;
+            let minDist = Number.MAX_VALUE; // set min to max dist
+
+            for (let i = 0; i < K; i++) {
+                let dist = computeDist(x,y,clusterX[i],clusterY[i]);
+
+                if (dist < minDist) {
+                    minCluster=i;
+                    minDist = dist;
+                }
+            }
+
+            return minCluster;
+        }
+
+        function computeCentroids() {
```

Comment:
This and other functions of the benchmark are duplicated. Would it be possible to move them to another ES module and then just `import` them around where needed?

Reply:
I wanted to reduce "noise" in the benchmark and therefore deemed the code duplication reasonable. I wouldn't keep the benchmark in the final version as the class doesn't need to contain a benchmark "test" as  it doesn't add any additional function tests.